setup.py

```diff
@@ -27,8 +27,12 @@
 #------------------------------------------------------------------------------
 
 from setuptools import setup
+import re
 
-import adal
+# setup.py shall not import adal
+__version__ = re.search(
+    "__version__[\s]*=[\s]*(.*)[\s]*",
```

Comment:
Originally I was hesitating to use those quotation-based regex (such as `'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]'` in requests package) because they will break when a version string happens to contain quotation marks inside (which is rare though). Now @yugangw-msft you raise a very good point that my proposed space-based regex can not handle trailing comments, which is quite normal. (BTW, I personally like to [use inline comments](http://stackoverflow.com/questions/6388187/what-is-the-proper-way-to-format-a-multi-line-dict-in-python/13761377#13761377) a lot.)

So I will use that quotation-based regex in my next commit. Stay tuned.